### PR TITLE
Hw18: done

### PR DESCRIPTION
**Pull request by Xiatelly**

### CODE REVIEW

File: hw18/src/SimpleTree.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        children.add(child);
+        return child;
+    }
+
+    public SimpleTree<T> addChild(T value){
+        SimpleTree<T> child = new SimpleTree<>(value);
+        return addChild(child);
+    }
+
+    public void iterate(Consumer<T> callback){
```
Comment (michael-trafficlabel) [inline hw18/src/SimpleTree.java:33]: Ensure that the value and children are properly handled, especially if null values might be encountered.

Comment (michael-trafficlabel) [inline hw18/src/SimpleTree.java:33]: it wasn`t metioned in task but please implement iterate method in the way to count sum of elements in tree

Comment (Xiatelly) [reply hw18/src/SimpleTree.java:33]: This was realized here, in jUnit test:
`AtomicInteger sumResult = new AtomicInteger();`
`tree0.iterate(sumResult::addAndGet);`

Comment (Xiatelly) [reply hw18/src/SimpleTree.java:33]: Processed.

**Pull request merged**